frontend/src/scenes/data-warehouse/editor/editorSidebarLogic.tsx

```diff
@@ -62,139 +57,273 @@ export const editorSidebarLogic = kea<editorSidebarLogicType>([
             ['selectSourceTable', 'toggleJoinTableModal'],
         ],
     }),
+    loaders(({ values }) => ({
+        folders: [
+            [] as DataWarehouseFolder[],
+            {
+                loadFolders: async () => {
+                    const response = await api.dataWarehouseFolders.list()
+                    return response.results
+                },
+                addFolder: async ({ name, parentId }) => {
+                    const folder = await api.dataWarehouseFolders.create({ name, parent: parentId })
+                    return values.folders.concat(folder)
+                },
+                deleteFolder: async ({ id }) => {
+                    await api.dataWarehouseFolders.delete(id)
+                    return values.folders.filter((folder) => folder.id !== id)
+                },
```

Comment:
I'm able to generate stacktraces when removing nested empty folders:
<img width="514" alt="image" src="https://github.com/user-attachments/assets/b5f7f27c-44ba-4cb4-96b9-1d04c39eb252" />